model/jpa/src/main/java/org/keycloak/models/jpa/entities/UserAttributeEntity.java

```diff
@@ -77,11 +80,20 @@ public void setName(String name) {
     }
 
     public String getValue() {
-        return value;
+        return value != null ? value : long_value;
```

Comment:
Would make sense to check whether both values are set to identify more easily an unexpected state where both values are set? I know it is unlike to happen but perhaps as an additional check. Not necessarily a Java assertion but perhaps an `IllegalStateException`.

Reply:
+1